app/services/org_nummers_generator.rb

```diff
@@ -0,0 +1,57 @@
+# Generate a valid Swedish OrganisationNummer (OrgNummer; OrgNr).
+#
+# It must be 10 digits and the last digit must be the valid Luhn checksum, per
+# https://sv.wikipedia.org/wiki/Organisationsnummer
+#
+
+require_relative './luhn_checksum'
+
+require 'set'
+
+class OrgNummersGenerator
+
+  MAX_NUM = 9999999999 unless defined?(MAX_NUM)
+  MIN_NUM = 0 unless defined?(MIN_NUM)
+
+  NONE_FOUND = nil  unless defined?(NONE_FOUND)
+
+
+  def self.generate(number_to_generate = 1)
+
+    results = Set.new
+
+    num_tries = 0
+    all_possible = MAX_NUM - MIN_NUM
+
+    # num_tries keeps use from trying forever.
+    # It is a somewhat arbitrary limit on how many times we can try.
+    # Even though each try is a Random number, so it is possible to get the same number more than once,
+    # this is a reasonble (if clunky) limit.
+    while results.count < number_to_generate && (num_tries < all_possible) do
+      number_to_generate.times do
+        org_num = generate_one
+        if results.add? org_num
+          results.add org_num
```

Comment:
I don't think this statement is needed, as if we get here the number has been added to the set by the previous statement.

Reply:
good catch. exactly right.